core/services/relay/delegate.go

```diff
@@ -36,19 +36,17 @@ type delegate struct {
 
 // NewDelegate creates a master relay delegate which manages "relays" which are OCR2 median reporting plugins
 // for various chains (evm and non-evm). nil Relayers will be disabled.
-func NewDelegate(ks keystore.Master, evmRelayer, solanaRelayer, terraRelayer types.Relayer) *delegate {
+func NewDelegate(ks keystore.Master) *delegate {
 	d := &delegate{
 		ks:       ks,
 		relayers: map[types.Network]types.Relayer{},
 	}
-	d.addRelayer(types.EVM, evmRelayer)
-	d.addRelayer(types.Solana, solanaRelayer)
-	d.addRelayer(types.Terra, terraRelayer)
 	return d
 }
 
-// addRelayer registers the relayer r, or a disabled placeholder if nil.
-func (d delegate) addRelayer(n types.Network, r types.Relayer) {
+// AddRelayer registers the relayer r, or a disabled placeholder if nil.
+// NOT THREAD SAFE
+func (d delegate) AddRelayer(n types.Network, r types.Relayer) {
```

Comment:
probably want to remove the disabled relayer with this change, it looks not needed anymore

Reply:
Done